File: leetcode/242. Valid Anagram/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.size() != t.size())
            return false;
        unordered_map<char, int> countS;
        unordered_map<char, int> countT;
        for (int i = 0; i < s.size(); i++)
        {
            countS[s[i]]++;
            countT[t[i]]++;
        }
        return countS == countT;
    }
};

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    Solution sol;

    string s, t;
    cin >> s >> t;

    cout << sol.isAnagram(s, t) << endl;
    return 0;
}

File: leetcode/242. Valid Anagram/solution.java
import java.util.Map;
import java.util.HashMap;
import java.util.Scanner;

class Solution {
    
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        
        Map<Character, Integer> countS = new HashMap<>();
        Map<Character, Integer> countT = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            countS.put(s.charAt(i), countS.getOrDefault(s.charAt(i), 0) + 1);
            countT.put(t.charAt(i), countT.getOrDefault(t.charAt(i), 0) + 1);
        }

        return countS.equals(countT);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        String t = sc.next();
        Solution solution = new Solution();
        System.out.println(solution.isAnagram(s, t) ? "true" : "false");
        sc.close();
    }
}